lib/vagrant/util/platform.rb

```diff
@@ -33,6 +33,22 @@ def cygwin?
           end
         end
 
+        def wsl?
+          return @_wsl if defined?(@_wsl)
+          @_wsl = -> {
+            platform = File.read('/proc/sys/kernel/osrelease')
+            return platform.include?("Microsoft")
+          }.call
+          return @_wsl
```

Comment:
Any reason for using a `proc` here? Should be able to reduce this and add an existence check as well:

```ruby
if @_wsl.nil?
  if File.exist?("/proc/sys/kernel/osrelease")
    platform = File.read("/proc/sys/kernel/osrelease")
    @_wsl = platform.include?("Microsoft")
  else
    @_wsl = false
  end
end
@_wsl